💄(jitsi-box:student) Display the secondary display as a floating draggable box

faydef opened the pull request:
choose a display(image or video) as your main display and the other as a seconday, therefore it is put in a floating box that you can drag to keep seeing without disturbing the main display, you can switch between diplay modes with a click of a button

faydef committed:
💄(jitsi-box:student) Display the secondary display as a floating draggable box

choose a display(image or video) as your main display and the other as a seconday, therefore it is put in a floating box that you can drag to keep seeing without disturbing the main display, you can switch between diplay modes with a click of a button

AladdinVR commented on box-ui/src/components/ts/FloatingBox.tsx:
```diff
@@ -0,0 +1,15 @@
+import Draggable from 'react-draggable';
+import '../css/FloatingBox.css';
+
+const FloatingBox = ({ children }: { children: any }) => {
+    return (
+        <div>
+            <Draggable>
+                <div className='pipWindow'>{children}</div>
+            </Draggable>
+        </div>
+    );
+};
```
```suggestion
const FloatingBox = ({ props }: FloatingBoxProps) => {
    return (
        <div>
            <Draggable>
                <div className='pipWindow'>{props.children}</div>
            </Draggable>
        </div>
    );
};
```

You should create a type for your props as it is done for other components in utils/props

djexvr replied:
If you only need children you can only add FunctionComponent to make typescript happy and then get children from props

djexvr commented on box-ui/src/components/css/FloatingBox.css:
```diff
@@ -0,0 +1,26 @@
+    font-size: 14px;
+    font-weight: 200;
+    line-height: 1.5;
+    color: rgb(248, 248, 249);
```
You don't need to modify the text css

djexvr commented on box-ui/src/components/css/FloatingBox.css:
```diff
@@ -0,0 +1,26 @@
+    color: rgb(248, 248, 249);
+    box-sizing: content-box;
+    box-shadow: 1px 3px 3px 0 rgb(0 0 0 / 20%), 1px 3px 15px 2px rgb(0 0 0 / 20%);
+    border-radius: 0.28571429rem;
```
This is very specific

faydef committed:
👔(studentMeeting:images) add the possibility to keep the original 50/50 display

add the possibility to keep the original display without a floating box
add better mor explicit buttons

djexvr commented on box-ui/src/components/css/FloatingBox.css:
```diff
@@ -0,0 +1,26 @@
+    outline: -webkit-focus-ring-color auto 1px;
+    outline-color: -webkit-focus-ring-color;
+    outline-style: auto;
+    outline-width: 1px;
```
You're repeating yourself the line 22 should suffice.

`-webkit-focus-ring-color` is only avalaible on safari you should add for firefox and chrome
```suggestion
    outline: -webkit-focus-ring-color auto 1px;
    outline: Highlight auto 1px;
```

djexvr replied:
You're repeating yourself the line 22 should suffice.

`-webkit-focus-ring-color` is only avalaible on safari you should add for firefox and chrome
```suggestion
    outline: -webkit-focus-ring-color auto 1px;
    outline: Highlight auto 1px;
```

faydef committed:
💫(studentMeeting:tuto) add toturial for the use of the floating window

add tutorial to explain the three buttons that were added to make sure the transitions work clearly

faydef committed:
✏️(studentMeeting: modify a forgotten F instead of the arrow icon

correct the icon of a button

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -191,22 +200,30 @@ const StudentMeeting: FunctionComponent = () => {
         }
     }
 
+    function ChangeView() {
+        setMiniImg((miniImg) => !miniImg);
+    }
+
+    function ChangeMinimize() {
+        setMinimize((minimize) => !minimize);
+    }
+
```
Do you really need a function for that can't you use it directly ?

djexvr commented on box-ui/src/components/ts/FloatingBox.tsx:
```diff
@@ -0,0 +1,15 @@
+
+const FloatingBox = ({ children }: { children: any }) => {
+    return (
+        <div>
```
Do you need this div. It's better to not add unnecessary div.

faydef committed:
💻(FloatingBox:utils.props) add props type to the component

add prop type to the component

djexvr commented on box-ui/src/utils/Props/index.d.ts:
```diff
@@ -67,3 +67,7 @@ export type FocusModeProps = {
     setDisplayFocus: React.Dispatch<React.SetStateAction<boolean>>;
 };
+
+export type FloatProps = {
```
You don't need to do this if you create it with FunctionComponent

djexvr commented:
I really think you should create a component for the Image Viewer and use it in StudentMeeting rather than recoding it in every floating or non-floating window it is really the opposite of the react philosophy and will make modifying the interface way more complicated and error-prone.

faydef committed:
💻(studentMeeting:UX) isolate image viewer in a component of its own
dquote>
dquote> make a component for the image viewer since it is used frequently

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -279,99 +280,35 @@ const StudentMeeting: FunctionComponent = () => {
+                            onclick={addCircle}
+                            addOn={circles}
+                            selectWindow={selectCoord}
+                        ></ImageViewer>
```
You can use an orphan tag `<ImageViewer ...... />`

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -420,49 +357,13 @@ const StudentMeeting: FunctionComponent = () => {
+                        img1={[img, '45vh', widthImgDouble.toString() + 'vh']}
+                        img2={[img, '45vh', ratioImgOriginal.toString() + 'vh']}
+                        selectWindow={selectCoord}
+                    ></ImageViewer>
```
Same as before use `<ImageViewer ..../>`

djexvr commented on box-ui/src/components/ts/FloatingBox.tsx:
```diff
@@ -5,11 +5,9 @@ import { FloatProps } from '../../utils/Props';
 
 const FloatingBox = ({ children }: FloatProps) => {
```
use `const FloatingBox: FunctionComponent = (props) =>` to be consistent with the other components.

faydef committed:
💄(StudentMeeting) switch icons for pop up

switch the icon for the pop up to the conventional one

faydef committed:
💻(StudentMeeting) improve readability

use orphan tags and clearer calls for onclick functions

djexvr commented:
Nice work well done 👍
